### PR TITLE
fix: balances and input address check

### PR DESCRIPTION
**Pull request by 0xArdy**
closes SUP-86
closes SUP-85

### CODE REVIEW

**Review by 0xtiti — DISMISSED** [review]
this looks good to! 🚀

**Review by 0xtiti — APPROVED** [review]

File: src/pages/[chain]/account/[account]/index.tsx [hunk 1]
```diff
@@ -85,6 +97,13 @@ const History = () => {
     }
   }, [getOrderedLogs, orderedLogs.length, isSuccess]);
 
+  useEffect(() => {
+    // if the user is disconnected, redirect to the home page
+    if (currentAddress?.toLowerCase() !== queryAddress?.toLowerCase()) {
```
Comment (turtlemoji) [inline src/pages/[chain]/account/[account]/index.tsx:102]: this is the best way to know if it is disconnected?

**Review by turtlemoji — APPROVED** [review]
pre-approved with comment

File: src/pages/[chain]/account/[account]/index.tsx (hunk 1 above)
Comment (0xArdy) [reply src/pages/[chain]/account/[account]/index.tsx:102]: yeap, `currentAddress` comes from `useAccount` (from wagmi)

**Pull request merged**